include/odp/api/spec/comp.h

```diff
@@ -0,0 +1,863 @@
+	 *  indicated by dict_len. if set to 0 and if support_dict ==
+	 *  ODP_SUPPORT_YES, then implementation will use dictionary length
+	 *  less than or equal to user input length in odp_comp_set_dict()
+	 *  and update used dictionary length at output of the call.
+	 *
+	 */
+	uint32_t dict_len;
+
+	/** Maximum compression level supported by implementation of this algo.
+	 * Indicates number of compression levels supported by implementation,
+	 *
+	 * where,
+	 *
+	 * 1 means fastest compression i.e. output produced at
+	 * best possible speed at the expense of compression quality, and
+	 *
+	 * max_level means best compression i.e.output produced is best possible
+	 * compressed content at the expense of speed.
+	 *
+	 * Example, if max_level = 4 , it means algorithm supports four levels
+	 * of compression from value 1 up to 4. User can set this value from
+	 * 1 (fastest compression) to 4 (best compression).
+	 * See RFC1950 for an example explanation to level.
+	 *
+	 * Value 0 mean implementation use its default value.
+	 *
+	 */
+	uint32_t max_level;
+
+	/** Supported hash algorithms */
+	odp_comp_hash_algos_t hash_algo;
+} odp_comp_alg_capability_t;
+
+/**
+ * Comp API dictionary type
+ * Consists of pointer to byte buffer. length of dictionary
+ * indicated by length parameter.
+ */
+typedef struct odp_comp_dict_t {
+	/** pointer to byte array */
+	uint8_t *buf;
+	/** length of the dictionary. */
+	uint32_t len;
+} odp_comp_dict_t;
+
+/**
+ * Comp API algorithm specific parameters
+ *
+ */
+typedef struct odp_comp_algo_param_t {
+	/**
+	 * Struct for defining deflate algorithm parameters.
+	 * Also initialized by other deflate based algorithms , ex. ZLIB
+	 */
+	struct comp_alg_def_param {
+		/**
+		 * compression level where
+		 * ODP_COMP_LEVEL_MIN <= level <= ODP_COMP_LEVEL_MAX
+		 */
+		odp_comp_level_t level;
+		/** huffman code to use */
+		odp_comp_huffman_code_t comp_code;
+	} deflate; /** deflate parameter */
+
+	/** Struct for defining zlib algorithm parameters.*/
+	struct comp_alg_zlib_param {
+			/** deflate algo params */
+			struct comp_alg_def_param def;
+	} zlib; /** zlib parameter */
+} odp_comp_alg_param_t;
+
+ /**
+ * Comp API session creation parameters
+ *
+ */
+typedef struct odp_comp_session_param_t {
+	/** Compress vs. Decompress operation */
+	odp_comp_op_t op;
+
+	/** Sync vs Async mode
+	 *
+	 * When mode = ODP_COMP_SYNC, odp_comp_xxx()
+	 * should be called.
+	 *
+	 * When mode = ODP_COMP_ASYNC, odp_comp_xxx_enq()
+	 * should be called.
+	 *
+	 * Use odp_comp_capability() for supported mode.
+	 *
+	 */
+	odp_comp_op_mode_t mode;
+
+	/** Compression algorithm
+	 *
+	 *  Use odp_comp_capability() for supported algorithms.
+	 */
+	odp_comp_alg_t comp_algo;
+
+	/** Hash algorithm
+	 *
+	 *  Use odp_comp_alg_capability() for supported hash algo for
+	 *  compression algo given as comp_alg. Implementation should not
+	 *  support hash only operation on data. output should always contain
+	 *  data + hash.
```

Comment:
"Implementation should not support hash only operation on data. output should always contain data + hash."

Maybe this can be removed since hash_algos are listed is per comp algo capability. Null comp algo can set all hash algo bits to zero (== hash not supported). If some HW supports null comp + hash algo, then that implementation may support hash only also in comp API.

Reply:
Typo: s/output/Output/

Are should and should not as used here intended to be SHOULD and SHOULD NOT in the sense of [RFC 2119](https://tools.ietf.org/html/rfc2119)?